common/src/main/java/org/opensearch/ml/common/contextmanager/ActivationRuleFactory.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.contextmanager;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Factory class for creating activation rules from configuration.
+ * Supports creating rules from configuration maps and combining multiple rules.
+ */
+@Log4j2
+public class ActivationRuleFactory {
+
+    public static final String TOKENS_EXCEED_KEY = "tokens_exceed";
+    public static final String MESSAGE_COUNT_EXCEED_KEY = "message_count_exceed";
+
+    /**
+     * Create activation rules from a configuration map.
+     * @param activationConfig the configuration map containing rule definitions
+     * @return a list of activation rules, or empty list if no valid rules found
+     */
+    public static List<ActivationRule> createRules(Map<String, Object> activationConfig) {
+        List<ActivationRule> rules = new ArrayList<>();
+
+        if (activationConfig == null || activationConfig.isEmpty()) {
+            return rules;
+        }
+
+        // Create tokens_exceed rule
+        if (activationConfig.containsKey(TOKENS_EXCEED_KEY)) {
+            try {
+                Object tokenValue = activationConfig.get(TOKENS_EXCEED_KEY);
+                int tokenThreshold = parseIntegerValue(tokenValue, TOKENS_EXCEED_KEY);
+                if (tokenThreshold > 0) {
+                    rules.add(new TokensExceedRule(tokenThreshold));
+                    log.debug("Created TokensExceedRule with threshold: {}", tokenThreshold);
+                } else {
+                    log.warn("Invalid token threshold value: {}. Must be positive integer.", tokenValue);
```

Comment:
Don't we need to throw exception here?

Reply:
most of the context managers processing has fallback, meaning help process the context at the background, if failed in context engineering, the worst case is the not applying any engineering on the context and keep the context as it is.

Reply:
added in the new commit